docs/index.md

```diff
@@ -31,64 +31,130 @@ You can easily integrate it with your existing tools, and feel lakeFS in action
+
+Today, recovering from these events relies on periodic backups that 1) may be out of date and 2) require sifting through data at the object level. This process is inefficient and can take hours, days, or in some cases, weeks to complete.
+Recovering data in deletion events becomes an instant one-line operation with lakeFS using the ability to restore any historical commit.
+
+Reverting your data lake back to previous version using our command-line tool is explained [here](https://docs.lakefs.io/reference/commands.html#lakectl-branch-reset).
+
+#### Data reprocessing and backfills
+
+Occasionally, we might need to reprocess historical data. This can be due to several reasons:
+
+* Implementing new logic.
+* Late arriving data that wasn’t included in previous analysis, and need to be backfilled after the fact.
 
-[Learn more](./usecases/production.md){:id="user-content-learn-more-dep"}
+This is tricky first and foremost because it often involves huge volumes of historical data. In addition, auditing requirements may necessitate keeping the old version of the data handy. 
+
+lakeFS allows you to manage the reprocess on an isolated branch before merging to ensure the reprocessed data is exposed atomically. It also allows you to easily access via a commit’s ID the different versions of reprocessed data.
+
+#### Cross-collection consistency guarantees
+
+Data engineers typically have to implement custom logic in scripts in order to guarantee two or more data assets are updated synchronously. The lakeFS merge operation from one branch into another removes the need to implement this logic yourself.
+
+Instead, make updates to the desired data assets on a branch, and then utilize a lakeFS merge to atomically expose the data to downstream consumers.
+
+To learn more about atomic cross-collection updates, [this video](https://www.youtube.com/watch?v=9OsjUvk5UJU) describes the concept in more detail.
+
+#### Troubleshooting production problems
+
+Data engineers are often asked to validate the data. A user might report inconsistencies, question the accuracy, or simply report it to be incorrect. Since the data continuously changes, it is challenging to understand its state at the time of the error
+
+The best way to investigate, therefore, is to have a snapshot of the data as close as possible to the time of the error.
+Once implementing a regular commit cadence in lakeFS, each commit represents an accessible historical snapshot of the data. When needed, a branch may be created from a commit ID to debug an issue in isolation.
+
+To learn more on how to access a specific historical commit in a repository, see our seminal post on [data reproducibility](https://lakefs.io/solving-data-reproducibility/).
+
+#### Establishing data quality guarantees
+
+The best way to deal with mistakes is to avoid them. A data source that is ingested into the lake introducing low-quality data should be blocked before exposure if possible.'
```

Comment:
```suggestion
The best way to deal with mistakes is to avoid them. A data source that is ingested into the lake introducing low-quality data should be blocked before exposure if possible.
```